framework/db/ActiveRecord.php

```diff
@@ -174,25 +174,47 @@ protected static function findByCondition($condition)
         $query = static::find();
 
         if (!ArrayHelper::isAssociative($condition)) {
-            // query by primary key
-            $primaryKey = static::primaryKey();
-            if (isset($primaryKey[0])) {
-                $pk = $primaryKey[0];
-                if (!empty($query->join) || !empty($query->joinWith)) {
-                    $pk = static::tableName() . '.' . $pk;
-                }
-                // if condition is scalar, search for a single primary key, if it is array, search for multiple primary key values
-                $condition = [$pk => is_array($condition) ? array_values($condition) : $condition];
-            } else {
-                throw new InvalidConfigException('"' . get_called_class() . '" must have a primary key.');
-            }
+            $pk = self::primaryKeyFieldWithAlias($query);
+            // if condition is scalar, search for a single primary key, if it is array, search for multiple primary key values
+            $condition = [$pk => is_array($condition) ? array_values($condition) : $condition];
+
         } elseif (is_array($condition)) {
             $condition = static::filterCondition($condition);
         }
 
         return $query->andWhere($condition);
     }
 
+    /**
+     * Returns field name of primary key with alias if defined
+     *
+     * @param ActiveQuery $query
+     * @return string
+     * @throws InvalidConfigException
+     */
+    private static function primaryKeyFieldWithAlias($query)
+    {
+        $primaryKey = static::primaryKey();
+
+        if (!isset($primaryKey[0])) {
+            throw new InvalidConfigException('"' . get_called_class() . '" must have a primary key.');
+        }
+
+        if (empty($query->join) && empty($query->joinWith)) {
+            return $primaryKey[0];
+        }
+
+        $tableOrAlias = static::tableName();
```

Comment:
In this conditions `$query->from` returns or null or single-element array. That is why we can just use
```suggestion
        $tableOrAlias = (!$query->from || isset($query->from[0])) ? static::tableName() : key($query->from);
```
without foreach loop, `array_keys` and even `is_string` funtions.